internal/ingest/ingest.go

```diff
@@ -825,15 +829,51 @@ func (ing *Ingester) ingestWorkerLogic(provider peer.ID) {
 			splitAtIndex = i
 			break
 		}
+		ctxIdStr := string(ai.ad.ContextID)
+		// This ad was deleted by a later remove. Push previous onto skips
+		// stack, and set latest skip.
+		if _, ok := rmCtxID[ctxIdStr]; ok {
```

Comment:
there's something odd about this skip stack:
* the first `skip` that will be put into `skips` will be '-1', the default value.
* that will then never matched in the subsequent loop through `i`, meaning that there seems to be at least 1 entry that isn't going to be skipped when it does match.